rdr_service/services/consent/files.py

```diff
@@ -28,76 +28,183 @@ def __init__(self, participant_id: int, storage_provider: GoogleCloudStorageProv
+        blob_wrapper = _ConsentBlobWrapper(blob)
+
+        if self._is_primary_consent(blob_wrapper):
+            return self._build_primary_consent(blob_wrapper)
+        elif self._is_cabor_consent(blob_wrapper):
+            return self._build_cabor_consent(blob_wrapper)
+        elif self._is_ehr_consent(blob_wrapper):
+            return self._build_ehr_consent(blob_wrapper)
+        elif self._is_gror_consent(blob_wrapper):
+            return self._build_gror_consent(blob_wrapper)
 
-    @abstractmethod
     def get_primary_consents(self) -> List['PrimaryConsentFile']:
+        return [
+            self._build_primary_consent(blob_wrapper)
+            for blob_wrapper in self.consent_blobs
+            if self._is_primary_consent(blob_wrapper)
+        ]
+
+    def get_cabor_consents(self) -> List['CaborConsentFile']:
+        return [
+            self._build_cabor_consent(blob_wrapper)
+            for blob_wrapper in self.consent_blobs
+            if self._is_cabor_consent(blob_wrapper)
+        ]
+
+    def get_ehr_consents(self) -> List['EhrConsentFile']:
+        return [
+            self._build_ehr_consent(blob_wrapper)
+            for blob_wrapper in self.consent_blobs
+            if self._is_ehr_consent(blob_wrapper)
+        ]
+
+    def get_gror_consents(self) -> List['GrorConsentFile']:
+        return [
+            self._build_gror_consent(blob_wrapper)
+            for blob_wrapper in self.consent_blobs
+            if self._is_gror_consent(blob_wrapper)
+        ]
+
+    @abstractmethod
+    def _is_primary_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> bool:
         ...
 
     @abstractmethod
-    def get_cabor_consents(self) -> List['CaborConsentFile']:
+    def _is_cabor_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> bool:
         ...
 
     @abstractmethod
-    def get_ehr_consents(self) -> List['EhrConsentFile']:
+    def _is_ehr_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> bool:
         ...
 
     @abstractmethod
-    def get_gror_consents(self) -> List['GrorConsentFile']:
+    def _is_gror_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> bool:
+        ...
+
+    @abstractmethod
+    def _build_primary_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> 'PrimaryConsentFile':
+        ...
+
+    @abstractmethod
+    def _build_cabor_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> 'CaborConsentFile':
+        ...
+
+    @abstractmethod
+    def _build_ehr_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> 'EhrConsentFile':
+        ...
+
+    @abstractmethod
+    def _build_gror_consent(self, blob_wrapper: '_ConsentBlobWrapper') -> 'GrorConsentFile':
         ...
 
     @abstractmethod
     def _get_source_bucket(self) -> str:
         ...
 
+    @abstractmethod
+    def _get_source_prefix(self) -> str:
+        ...
+
 
 class VibrentConsentFactory(ConsentFileAbstractFactory):
     CABOR_TEXT = (
```

Comment:
For future consideration: I think we may want to "share" these text identifers between Vibrent/CE ConsentFactory;  I think they both have to be using the same IRB-approved text.

Reply:
I think that's a great idea. There may be some other functionality that could me moved into the base class too when we work on the CE factory.